pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -141,66 +142,86 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
+	logger = logger.WithValues("username", userSignup.Spec.Username)
 
-	reqLogger = reqLogger.WithValues("username", instance.Spec.Username)
-	if instance.Labels[toolchainv1alpha1.UserSignupStateLabelKey] == "" {
-		if err := r.setStateLabel(reqLogger, instance, toolchainv1alpha1.UserSignupStateLabelValueNotReady); err != nil {
+	if userSignup.Labels[toolchainv1alpha1.UserSignupStateLabelKey] == "" {
+		if err := r.setStateLabel(logger, userSignup, toolchainv1alpha1.UserSignupStateLabelValueNotReady); err != nil {
 			return reconcile.Result{}, err
 		}
 	}
 
-	banned, err := r.isUserBanned(reqLogger, instance)
+	// (migration for 'complete' usersignups only) set the UserSignupActivationCounterAnnotationKey if it is missing
+	// To be removed once the operator was deployed in production, as per https://issues.redhat.com/browse/CRT-1036
+	if condition.IsTrueWithReason(userSignup.Status.Conditions, toolchainv1alpha1.UserSignupComplete, "") {
+		// obtain a lock to avoid race condition with the counter initializations
+		counter.InitializationMutex.Lock()
+		defer counter.InitializationMutex.Unlock()
```

Comment:
as discussed in the slack channel - we don't need it https://coreos.slack.com/archives/CHK0J6HT6/p1618836017462800?thread_ts=1618828935.454500&cid=CHK0J6HT6

Reply:
yes, this was reverted in https://github.com/codeready-toolchain/host-operator/pull/401/commits/b9a363953f5e17da803890ada2eae6c0615da96e